app/src/main/java/com/codesoom/assignment/controllers/TaskController.java

```diff
@@ -0,0 +1,103 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.exception.ParameterEmptyException;
+import com.codesoom.assignment.exception.TaskIdNotFoundException;
+import com.codesoom.assignment.model.Task;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+@RestController
+@RequestMapping("/tasks")
+@CrossOrigin
+public class TaskController {
+    private List<Task> tasks = new ArrayList<>();
+
+    @GetMapping
+    public List<Task> taskList() {
+        return tasks;
+    }
+
+    @GetMapping("/{id}")
+    public Task taskDetail(@PathVariable Long id) {
+        Task task = findTask(id);
+        return task;
+    }
+
+    @PostMapping
+    public ResponseEntity<Task> taskCreate(@RequestBody Task task) throws Throwable {
+        task.setId(generatedId());
+        tasks.add(task);
+        URI uri = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .path("/{id}")
+                .buildAndExpand(task.getId())
+                .toUri();
+        return new ResponseEntity<Task>(task,  getHttpLocationHeaders(uri), HttpStatus.CREATED);
+    }
+
+    private HttpHeaders getHttpLocationHeaders(URI uri) {
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.setLocation(uri);
+        return httpHeaders;
+    }
+
+    @RequestMapping(path = "/{id}", method = {RequestMethod.PATCH, RequestMethod.PUT})
+    public Task taskModify(@PathVariable Long id, @RequestBody Task task) {
+        validateParameter(id, task);
+        Task findTask = findTask(id);
+        int findTaskIndex = tasks.indexOf(findTask);
+        task.setId(findTask.getId());
+        tasks.set(findTaskIndex, task);
+        return task;
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity taskDelete(@PathVariable Long id) {
```

Comment:
`@ResponseStatus(HttpStatus.NO_CONTENT)`라고 어노테이션으로 처리하고 반환 타입을 `void`로 해도 괜찮을 것 같아요
```suggestion
    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void taskDelete(@PathVariable Long id) {
```

Reply:
@ReponseStatus로 반환 하는게 더 나은 것 같습니다~